pkg/scaffold/cmd_test.go

```diff
@@ -84,6 +86,22 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Become the leader before proceeding
+	leader.Become(context.TODO(), "app-operator-lock")
+
+	// Create file for readiness probe
+	f, err := os.Create("/tmp/operator-sdk-leader")
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+	err = f.Close()
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+	defer os.Remove(f.Name())
```

Comment:
@mhrivnak What do you think about providing a helper to abstract away Setting and Unsetting the Readiness check of an operator?
e.g `SetReady()` and `UnsetReady()`. That way the user can unset the Readiness if necessary based on the operator setup logic.
Plus it's a bit cleaner than having to create the file in `main.go` e.g:
https://github.com/coreos/etcd-operator/blob/master/pkg/util/probe/readyz.go#L31-L35
https://github.com/coreos/etcd-operator/blob/master/pkg/controller/informer.go#L51

And as for the actual probe, do you have any thoughts on using a CMD probe(stat file) vs an HTTP Get probe.
https://github.com/coreos/etcd-operator/blob/master/cmd/operator/main.go#L105

I've seen a few other operators use the HTTP probe, although I'm not sure that's a standard for upstream components like the scheduler, apiserver etc.
https://github.com/operator-framework/operator-metering/blob/master/manifests/deploy/generic/helm-operator/metering-operator-deployment.yaml#L86-L89
https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/statefulset.go#L512-L516

Reply:
I love the helper idea. I moved that part to a new package; see what you think.

Reply:
As for the probe type, http is useful if you want to dynamically determine readiness at the time a probe runs. For our current use case, I think it's overkill to have an entire web handler, and probably listen on yet another socket, just to return a boolean value that's being stored as a singleton in memory. It's much simpler to save that boolean state to the filesystem (by creating or deleting a file at a known path) and let k8s access that state on-disk through normal POSIX means.

You probably saw this, but for completeness, the main example for liveness/readiness probes uses the same file-existence technique: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#define-a-liveness-command

That said, I made an interface to set and unset the state so that if a future operator dev has a need for an http-based readiness probe, or any other kind besides the simple file-based probe, that can be facilitated.